src/frontend/src/tests/lib/utils/tokens.utils.spec.ts

```diff
@@ -368,3 +369,65 @@ describe('calculateTokenUsdBalance', () => {
 		expect(result).toEqual(0);
 	});
 });
+
+describe('sumMainnetTokensUsdBalance', () => {
+	const mockUsdValue = usdValue as MockedFunction<typeof usdValue>;
+
+	beforeEach(() => {
+		vi.resetAllMocks();
+
+		mockUsdValue.mockImplementation(
+			({ balance, exchangeRate }) => Number(balance ?? 0) * exchangeRate
+		);
+	});
+
+	it('should correctly calculate USD total balance for the list of mainnet and testnet tokens', () => {
+		const balances = {
+			...$balances,
+			[BTC_TESTNET_TOKEN.id]: { data: bn3, certified }
+		};
+		const tokens = [...$tokens, BTC_TESTNET_TOKEN];
+
+		const result = sumMainnetTokensUsdBalance({ $tokens: tokens, $balances: balances, $exchanges });
+		expect(result).toEqual(bn1.toNumber() + bn2.toNumber() + bn3.toNumber());
+	});
+
+	it('should return 0 if no mainnet tokens are in the list', () => {
+		const balances = {
+			...$balances,
+			[BTC_TESTNET_TOKEN.id]: { data: bn2, certified }
+		};
+		const tokens = [BTC_TESTNET_TOKEN];
+
+		const result = sumMainnetTokensUsdBalance({ $tokens: tokens, $balances: balances, $exchanges });
+		expect(result).toEqual(0);
+	});
+
+	it('should return 0 if all token balances are 0', () => {
+		const balances = {
+			[ICP_TOKEN.id]: { data: ZERO, certified },
+			[BTC_MAINNET_TOKEN.id]: { data: ZERO, certified },
+			[ETHEREUM_TOKEN.id]: { data: ZERO, certified },
+			[BTC_TESTNET_TOKEN.id]: { data: ZERO, certified }
+		};
+		const tokens = [...$tokens, BTC_TESTNET_TOKEN];
+
+		const result = sumMainnetTokensUsdBalance({ $tokens: tokens, $balances: balances, $exchanges });
+		expect(result).toEqual(0);
+	});
+
+	it('should return 0 if no tokens are provided', () => {
+		const result = sumMainnetTokensUsdBalance({ $tokens: [], $balances, $exchanges });
+		expect(result).toEqual(0);
+	});
+
+	it('should return 0 if exchange rate is not available', () => {
```

Comment:
i think the last two tests are not needed, since they refer to testing `calculateTokenUsdBalance` basically, right?

Reply:
You're right, no need to test these 2 cases for this function. Removed.